modules/renter/hostdb/hostweight.go

```diff
@@ -31,113 +33,115 @@ var (
 
 // calculateHostWeight returns the weight of a host according to the settings of
 // the host database entry. Currently, only the price is considered.
-func calculateHostWeight(currentHeight types.BlockHeight, entry hostEntry) (weight types.Currency) {
-	// Prices tiered as follows:
-	//    - the storage price is presented as 'per block per byte'
-	//    - the contract price is presented as a flat rate
-	//    - the upload bandwidth price is per byte
-	//    - the download bandwidth price is per byte
-	//
-	// The hostdb will naively assume the following for now:
-	//    - each contract covers 6 weeks of storage (default is 12 weeks, but
-	//      renewals occur at midpoint) - 6048 blocks - and 10GB of storage.
-	//    - uploads happen once per 12 weeks (average lifetime of a file is 12 weeks)
-	//    - downloads happen once per 6 weeks (files are on average downloaded twice throughout lifetime)
-	//
-	// In the future, the renter should be able to track average user behavior
-	// and adjust accordingly. This flexibility will be added later.
-	adjustedContractPrice := entry.ContractPrice.Div64(6048).Div64(10e9) // Adjust contract price to match 10GB for 6 weeks.
-	adjustedUploadPrice := entry.UploadBandwidthPrice.Div64(24192)       // Adjust upload price to match a single upload over 24 weeks.
-	adjustedDownloadPrice := entry.DownloadBandwidthPrice.Div64(12096)   // Adjust download price to match one download over 12 weeks.
-	siafundFee := adjustedContractPrice.Add(adjustedUploadPrice).Add(adjustedDownloadPrice).Add(entry.Collateral).MulTax()
-	totalPrice := entry.StoragePrice.Add(adjustedContractPrice).Add(adjustedUploadPrice).Add(adjustedDownloadPrice).Add(siafundFee)
+func (hdb *HostDB) calculateHostWeight() hosttree.WeightFunc {
+	return func(entry modules.HostDBEntry) types.Currency {
```

Comment:
I don't like the tabbing here, would prefer to see the weight func defined separately, then `return weightFunc` or whatever. I don't think it makes sense to be inline here, it's a massive amount of code.